lisa/sut_orchestrator/azure/features.py

```diff
@@ -1897,6 +1897,20 @@ def remove_data_disk(self, names: Optional[List[str]] = None) -> None:
         self._node.capability.disk.data_disk_count -= len(names)
         self._node.close()
 
+    def get_resource_disks(self) -> List[str]:
+        resource_disk_mount_point = self.get_resource_disk_mount_point()
+        resource_disks = []
+        try:
+            resourcedisk = self._node.features[Disk].get_partition_with_mount_point(
+                resource_disk_mount_point
+            )
+            if resourcedisk:
+                resource_disks = [resourcedisk.name]
+        except AssertionError:
```

Comment:
Is there a way not check by the error? The error can be happened by other reasons, the eating code may hide other problems.

Reply:
Modified the logic to detect the issue at lower level to avoid this.